backends/xsmm/ceed-xsmm.h

```diff
@@ -15,14 +15,19 @@
 // testbed platforms, in support of the nation's exascale computing imperative.
 
 #include <ceed-backend.h>
+#include <ceed-hash.h>
 #include <libxsmm.h>
 #include <string.h>
 #include <math.h>
 
+// Instantiate khash structs and methods
+CeedHashIJKLMInit(m32, char)
```

Comment:
Why map to char instead of to the kernel itself? Then you can avoid shuffling indexes.

Reply:
updated